airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,469 @@
+    instances to multiple mesos workers.
+
+    Apache Mesos is a distributed systems kernel which abstracts
+    CPU, memory, storage, and other compute resources away from
+    machines (physical or virtual), enabling fault-tolerant and
+    elastic distributed systems to easily be built and run effectively.
+    See http://mesos.apache.org/
+    """
+
+    class MesosFramework(threading.Thread):
+        """MesosFramework class to start the threading"""
+
+        def __init__(self, client):
+            threading.Thread.__init__(self)
+            self.client = client
+            self.stop = False
+
+        def run(self):
+            try:
+                self.client.register()
+            except KeyboardInterrupt:
+                self.log.info("Stop requested by user, stopping framework....")
+
+    def __init__(self):
+        super().__init__()
+        self.commands_to_run = []
+        self.task_queue = Queue()
+        self.result_queue = Queue()
+        self.driver = None
+        self.client = None
+        self.th = None  # pylint: disable=invalid-name
+
+    @provide_session
+    def start(self, session=None):
+        """Setup and start routine to connect with the mesos master"""
+        if not conf.get('mesos', 'MASTER'):
+            self.log.error("Expecting mesos master URL for mesos executor")
+            raise AirflowException("mesos.master not provided for mesos executor")
+
+        master = conf.get('mesos', 'MASTER')
+
+        framework_name = get_framework_name()
+        framework_id = None
+
+        task_cpu = conf.get('mesos', 'TASK_CPU', fallback=1)
+        task_memory = conf.get('mesos', 'TASK_MEMORY', fallback=256)
+
+        if conf.getboolean('mesos', 'CHECKPOINT'):
+            framework_checkpoint = True
+
+            if conf.get('mesos', 'FAILOVER_TIMEOUT'):
+                # Import here to work around a circular import error
+                from airflow.models import Connection
+
+                # Query the database to get the ID of the Mesos Framework, if available.
+                conn_id = FRAMEWORK_CONNID_PREFIX + framework_name
+                connection = session.query(Connection).filter_by(conn_id=conn_id).first()
+                if connection is not None:
+                    # Set the Framework ID to let the scheduler reconnect
+                    # with running tasks.
+                    framework_id = connection.extra
+
+                # Set Timeout in the case of a mesos master leader change
+                framework_failover_timeout = conf.getint(
+                    'mesos', 'FAILOVER_TIMEOUT'
+                )
+
+        else:
+            framework_checkpoint = False
+
+        self.log.info(
+            'MesosFramework master : %s, name : %s, cpu : %s, mem : %s, checkpoint : %s, id : %s',
+            master, framework_name,
+            str(task_cpu), str(task_memory), str(framework_checkpoint), str(framework_id)
```

Comment:
Are those string casts necessary?

Reply:
Actually, no. Just a piece of old code. I already changed it.